apps/wallet/src/ui/app/components/transactions-card/Transaction.tsx

```diff
@@ -0,0 +1,204 @@
+            )?.recipientAddress;
+        const receiverAddr =
+            eventsSummary &&
+            eventsSummary.find(
+                ({ receiverAddress }) => receiverAddress !== address
+            )?.receiverAddress;
+
+        return (
+            receiverAddr ??
+            tranferObjectRecipientAddress ??
+            certificate.data.sender
+        );
+    }, [address, amountByRecipient, certificate.data.sender, eventsSummary]);
+
+    // receiving transaction from move contract sometimes has the sender as the receiver
+    // so we need to check if the sender is the same as the address
+    const isSender = address === certificate.data.sender;
+
+    const receiverAddress = useMiddleEllipsis(
+        recipientAddress || '',
+        TRUNCATE_MAX_LENGTH,
+        TRUNCATE_PREFIX_LENGTH
+    );
+
+    const txnLabel = useMemo(() => {
+        const moveCallTxn = getMoveCallTransaction(
+            certificate.data.transactions[0]
+        );
+        if (txnKind === 'Call')
+            return moveCallTxn?.function.replace(/_/g, ' ') || 'Call';
+        if (txnKind === 'ChangeEpoch') return txnKind;
+        return recipientAddress;
+    }, [certificate.data.transactions, recipientAddress, txnKind]);
+
+    const txnDate = useMemo(() => {
+        return txn?.timestamp_ms
+            ? formatDate(txn.timestamp_ms, ['month', 'day', 'hour', 'minute'])
+            : false;
+    }, [txn]);
+
+    const error = useMemo(() => getExecutionStatusError(txn), [txn]);
+    const isSuiTransfer =
+        txnKind === 'PaySui' ||
+        txnKind === 'TransferSui' ||
+        txnKind === 'PayAllSui' ||
+        txnKind === 'Pay';
+
+    const label = useMemo(() => {
+        return isSuiTransfer || txnKind === 'TransferObject'
+            ? isSender
+                ? 'To'
+                : 'From'
+            : 'Action';
+    }, [isSender, isSuiTransfer, txnKind]);
+
+    return (
+        <Link
+            to={`/receipt?${new URLSearchParams({
+                txdigest: txn.certificate.transactionDigest,
+            }).toString()}`}
+            className="flex items-center w-full flex-col gap-2 py-4 no-underline"
+        >
+            <div className="flex items-start w-full justify-between gap-3">
+                <div className="w-7.5">
+                    <TxnIcon
+                        txnFailed={executionStatus !== 'success' || !!error}
+                        isSender={isSender}
+                    />
+                </div>
+                <div className="flex flex-col w-full gap-1.5">
+                    {error ? (
+                        <div className="flex flex-col w-full gap-1.5">
+                            <Text color="gray-90" weight="semibold">
+                                Transaction Failed
+                            </Text>
+                            <div className="p-2">
+                                <Alert mode="warning">
```

Comment:
dont know if we want to use Alert here based on the design

<img width="362" alt="image" src="https://user-images.githubusercontent.com/122397493/214244280-f0386cd2-611b-405b-82ef-e25a5a4f20ac.png">

Reply:
reverted